nixos/modules/system/boot/systemd.nix

```diff
@@ -628,7 +635,13 @@ in
 
     environment.systemPackages = [ systemd ];
 
-    environment.etc = {
+    environment.etc = let
+      generators = (fold (p: attrs: let path = "${p}/lib/systemd/system-generators";
+                                    in attrs // (mapAttrs' (name: _: nameValuePair name "${path}/${toString name}")
+                                                           (builtins.readDir "${path}")))
```

Comment:
It likely does. I wasn't aware that this may cause issues. I'll refactor this PR accordingly.